### PR TITLE
[FIX] update estimator checks for sklearn 1.6

### LINKED ISSUE #4924: Test failures with sklearn 1.6
Only runs that pass are those with the oldest version of sklearn we support: all the others were run with 1.6

https://github.com/nilearn/nilearn/actions/runs/12241078484

Failures are due to some of the estimator checks

```
FAILED nilearn/maskers/tests/test_base_masker.py::test_check_estimator_invalid[estimator4-check4-check_positive_only_tag_during_fit]
FAILED nilearn/decomposition/tests/test_base.py::test_check_estimator[estimator9-check9-check_estimator_sparse_array] - AssertionError: Estimator _BaseDecomposition doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/maskers/tests/test_multi_nifti_masker.py::test_check_estimator[estimator7-check7-check_estimator_sparse_array] - AssertionError: Estimator MultiNiftiMasker doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/maskers/tests/test_multi_nifti_masker.py::test_check_estimator[estimator19-check19-check_estimator_sparse_array] - AssertionError: Estimator NiftiMasker doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/regions/tests/test_hierarchical_kmeans_clustering.py::test_check_estimator_invalid[estimator2-check2-check_positive_only_tag_during_fit]
````

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4924 

Changes proposed in this pull request:

- update list of estimator checks for sklearn 1.6

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@hndgzkn if you are interested on checking this small one.

File: nilearn/maskers/tests/test_base_masker.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 if compare_version(sklearn_version, "<", "1.5.0"):
     extra_valid_checks.append("check_estimator_sparse_data")
 
-if compare_version(sklearn_version, ">", "1.6"):
+if compare_version(sklearn_version, ">", "1.5.2"):
```
Comment (hndgzkn) [inline nilearn/maskers/tests/test_base_masker.py:34]: Could it be `sklearn_version, ">=", "1.6"` instead?

**Review by hndgzkn — APPROVED** [review]
There is also this broken link https://scikit-learn.org/stable/modules/model_evaluation.html#common-cases-predefined-values .

A part from that it's ok I think.

**Comment by Remi-Gau** [comment]
merging as this is pretty much blocking all other PRs

File: nilearn/maskers/tests/test_base_masker.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/maskers/tests/test_base_masker.py:34]: oops sorry, merged before refreshing the page

I think I tried this but you get into some 'weird' things with dev / pre release versions:

```
In [9]: compare_version("1.6.0", ">=", "1.6")
Out[9]: True

In [10]: compare_version("1.6.dev0", ">=", "1.6")
Out[10]: False
```

that's why things were working... until 1.6 was out....

probably we could improve `compare_version`

**Comment by Remi-Gau** [comment]
> There is also this broken link

- I think I am fixing this as part of https://github.com/nilearn/nilearn/pull/4911#pullrequestreview-2491937802

**Pull request merged**